internal/mcfg/mcfg.go

```diff
@@ -0,0 +1,82 @@
+package mcfg
+
+import (
+	"bytes"
+	"embed"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"text/template"
+)
+
+const (
+	kernelModuleImageFilepath = "/var/lib/image_file_day1.tar"
+	workerConfigFilepath      = "/var/lib/kmm_day1_config.yaml"
+)
+
+var (
+	//go:embed scripts/pull-image.sh
+	scriptPullImage string
+
+	//go:embed scripts/replace-kernel-module.sh
+	scriptReplaceKmod string
+
+	//go:embed scripts/wait-for-dispatcher.sh
+	scriptWaitForNetworkDispatcher string
+
+	//go:embed templates
+	templateFS embed.FS
+
+	ignitionTemplate = template.Must(
+		template.ParseFS(templateFS, "templates/ignition.gotmpl"),
+	)
+)
+
+//go:generate mockgen -source=mcfg.go -package=mcfg -destination=mock_mcfg.go
+
+type MCFG interface {
```

Comment:
```suggestion
type MC interface {
```

MC is the common acronym for MachineConfig in OCP.

Reply:
What is the difference between MachineConfig and MachineConfiguration?